### ISSUE TITLE
[BUG] Wrong Decimal/floating numbers when Excel->CSV

### ISSUE THREAD

**Issue by Prid13** [jqnatividad/qsv#516]
### Problem
Noticed this super weird occurrence when converting Excel files to CSV via `qsv excel file.xls`:
if a row value is a **decimal** with certain number of fractional digits, the fraction becomes imprecise:

![image](https://user-images.githubusercontent.com/27897747/193198524-7782041b-b380-48b5-9a7d-c39e05b8aaf3.png)

Am I doing something wrong? I tried using `--dates-whitelist none`, but to no avail. Any solutions? :)

**Comment by jqnatividad** [comment]
Hi @Prid13 ,
I was able to reproduce your bug by saving the file in Excel 97-2004 format (xls). 
Interestingly, it works as expected using the xlsx format.

For now, use xlsx  as the bug is most likely with the calamine library, for which I do have a fork that I maintain for qsv.

**Comment by Prid13** [comment]
You're absolutely right, the files I'm working with are indeed in the old `.xls` format.

I can't change to xlsx format since the files are "reports" sent every day via email, and converting them would take away from the automatization flow.

Do you know of any good CLI tools for seamlessly converting `xlsx` to `xls`? :)

**Comment by jqnatividad** [comment]
You may want to try https://pypi.org/project/xls-to-csv/

I'll ping you once my calamine fork is fixed and qsv handles xls floats properly.

**Comment by Prid13** [comment]
Thanks 😇⭐

After having searched for quite a bit, the only _dependency-free_ CLI tool I could find that supports `xls` files is this one: https://github.com/r-ook/xl2csv

It's a Python script that uses the `xlrd` library, packaged and released as a standalone `.exe` file. It's a bit slow, but it actually does the job :)

The only downside I found is that it converted all integers to float values, e.g. `14` -> `14.0`. After searching for a bit, I learned that `.xls` only stores floating point values, while `.xlsx` has support for integers as well. But hopefully that won't be an issue for my use case.

Also found this amazing tool: https://github.com/tobya/DocTo -- but then I discovered that it requires the office programs to be installed on the host computer :/

Will be looking forward to that update, but no rush at all 😇
